lib/api/href.rb

```diff
@@ -0,0 +1,107 @@
+module Api
+  class Href
+    # @param href [String] the href string
+    #
+    #   Supports the following forms:
+    #   - http://www.example.com/api
+    #   - http://www.example.com/api/collection
+    #   - http://www.example.com/api/collection/:cid
+    #   - http://www.example.com/api/collection/:cid/subcollection
+    #   - http://www.example.com/api/collection/:cid/subcollection/:sid
+    #
+    #   Hrefs can also be supplied using just the path,
+    #   i.e. /api/collection/:cid. The preceding /, /api may also be
+    #   omitted, and all examples may be suffixed with a /.
+    def initialize(href)
+      @href = href
+    end
+
+    # @return [String, nil] the name of the subcollection from
+    #   the href if there is one, otherwise returns the name of the
+    #   collection. If neither are present, returns nil
+    def resource
```

Comment:
Why this change ?  This is confusing for me, I prefer the current subject. For the API a resource is the managed CI, i.e. "provider/1" or "provider/1/vms/20",  not the targeted collection "provider" or subcollection "vms".